cli/src/maintenance.rs

```diff
@@ -798,6 +829,58 @@ impl SolidoState {
         None
     }
 
+    /// Unstake from active validators in order to rebalance validators.
+    pub fn try_unstake_from_active_validators(&self) -> Option<(Instruction, MaintenanceOutput)> {
+        // Return None if there's no active validator to unstake from.
+        self.solido.validators.iter_active().next()?;
+
+        // Get the target for each validator. Undelegated Lamports can be
+        // sent when staking with validators.
+        let targets = lido::balance::get_target_balance(Lamports(0), &self.solido.validators)
+            .expect("Failed to compute target balance.");
+        let unstake_amounts =
+            lido::balance::get_unstake_to_rebalance(&self.solido.validators, &targets);
+
+        let validator_index = lido::balance::get_unstake_validator_index(
+            &self.solido.validators,
+            &targets,
+            &unstake_amounts,
+            SolidoState::UNBALANCE_THRESHOLD,
+        )?;
+        let validator = &self.solido.validators.entries[validator_index];
+        let stake_account = &self.validator_stake_accounts[validator_index][0];
+
+        // Get the maximum that can be unstaked from the stake account.
+        let amount = unstake_amounts[validator_index].min(stake_account.1.balance.total());
+
+        let expected_balance = (validator.entry.effective_stake_balance() - amount)
+            .expect("amount is always <= balance");
+        let amount = if expected_balance > MINIMUM_STAKE_ACCOUNT_BALANCE {
+            Some(amount)
+        } else {
+            let min_amount = (validator.entry.effective_stake_balance()
```

Comment:
This should be the amount in the first stake account, not the total effective balance.